packages/lib/modules/hooks/useHook.ts

```diff
@@ -1,18 +1,30 @@
-import { Pool } from '../pool/PoolProvider'
+import {
+  GqlChain,
+  GqlHook,
+  GqlPoolTokenDetail,
+} from '@repo/lib/shared/services/api/generated/graphql'
 import { useHooks } from './HooksProvider'
 import { getChainId } from '@repo/lib/config/app.config'
 
-export function useHook(pool: Pool) {
+export function useHook({
+  hook,
+  poolTokens,
+  chain,
+}: {
```

Comment:
Please revert to just pass in the pool. It's more verbose in all the components you need to use this hook or the hook tag component otherwise.

Reply:
There are different pool types in pools vs pool query, that's why decided to pass just properties

Reply:
Yeah, like I said, can you not just use the type `Pool | PoolListItem`?

Reply:
fixed